src/app/clusters/thread-border-router-management-server/thread-border-router-management-server.cpp

```diff
@@ -177,55 +175,39 @@ void ServerInstance::InvokeCommand(HandlerContext & ctxt)
     }
 }
 
-CHIP_ERROR ServerInstance::ReadFeatureMap(BitFlags<Feature> & featureMap)
+void ServerInstance::ReadFeatureMap(BitFlags<Feature> & outFeatureMap)
 {
-    bool panChangeSupported;
-    ReturnErrorOnFailure(mDelegate->GetPanChangeSupported(panChangeSupported));
-    if (panChangeSupported)
+    if (mDelegate->GetPanChangeSupported())
     {
-        featureMap.Set(Feature::kPANChange);
+        outFeatureMap.Set(Feature::kPANChange);
     }
-    return CHIP_NO_ERROR;
 }
 
-CHIP_ERROR ServerInstance::ReadBorderRouterName(MutableCharSpan & borderRouterName)
+CHIP_ERROR ServerInstance::ReadBorderRouterName(MutableCharSpan & outBorderRouterName)
 {
-    ReturnErrorOnFailure(mDelegate->GetBorderRouterName(borderRouterName));
-    ReturnErrorCodeIf(borderRouterName.size() > kBorderRouterNameMaxLength, CHIP_IM_GLOBAL_STATUS(ConstraintError));
+    mDelegate->GetBorderRouterName(outBorderRouterName);
+    VerifyOrReturnValue(outBorderRouterName.size() <= kBorderRouterNameMaxLength, CHIP_IM_GLOBAL_STATUS(Failure));
     return CHIP_NO_ERROR;
 }
 
-CHIP_ERROR ServerInstance::ReadBorderAgentID(MutableByteSpan & borderAgentId)
+CHIP_ERROR ServerInstance::ReadBorderAgentID(MutableByteSpan & outBorderAgentId)
 {
-    ReturnErrorOnFailure(mDelegate->GetBorderAgentId(borderAgentId));
-    ReturnErrorCodeIf(borderAgentId.size() != kBorderAgentIdLength, CHIP_IM_GLOBAL_STATUS(ConstraintError));
+    VerifyOrReturnValue((mDelegate->GetBorderAgentId(outBorderAgentId) == CHIP_NO_ERROR) &&
+                            (outBorderAgentId.size() == kBorderAgentIdLength),
+                        CHIP_IM_GLOBAL_STATUS(Failure));
     return CHIP_NO_ERROR;
 }
 
-CHIP_ERROR ServerInstance::ReadThreadVersion(uint16_t & threadVersion)
+Optional<uint64_t> ServerInstance::ReadActiveDatasetTimestamp()
 {
-    return mDelegate->GetThreadVersion(threadVersion);
-}
-
-CHIP_ERROR ServerInstance::ReadInterfaceEnabled(bool & interfaceEnabled)
-{
-    return mDelegate->GetInterfaceEnabled(interfaceEnabled);
-}
-
-CHIP_ERROR ServerInstance::ReadActiveDatasetTimestamp(Optional<uint64_t> & activeDatasetTimestamp)
-{
-    uint64_t activeDatasetTimestampValue;
+    uint64_t activeDatasetTimestampValue = 0;
     Thread::OperationalDataset activeDataset;
-    if (mDelegate->GetDataset(activeDataset, Delegate::DatasetType::kActive) == CHIP_NO_ERROR &&
-        activeDataset.GetActiveTimestamp(activeDatasetTimestampValue) == CHIP_NO_ERROR)
-    {
-        activeDatasetTimestamp.SetValue(activeDatasetTimestampValue);
-    }
-    else
+    if ((mDelegate->GetDataset(activeDataset, Delegate::DatasetType::kActive) == CHIP_NO_ERROR) &&
+        (activeDataset.GetActiveTimestamp(activeDatasetTimestampValue) == CHIP_NO_ERROR))
```

Comment:
There needs to be a notification when the dataset becomes active.

I don't see a path to ever mark the ActiveDatasetTimestamp attribute as dirty reactively to the Thread stack state. It should be so.

Reply:
Added the attribute change callback to report the attribute changes